experiment/simulation/js/js/materials.js

```diff
@@ -0,0 +1,39 @@
+import * as THREE from "https://threejsfundamentals.org/threejs/resources/threejs/r115/build/three.module.js";
+
+function vertexShader() {
+    return `varying vec3 vUv; 
+      
+                  void main() {
+                    vUv = position; 
+      
+                    vec4 modelViewPosition = modelViewMatrix * vec4(position, 1.0);
+                    gl_Position = projectionMatrix * modelViewPosition; 
+                  }`;
+}
+
+function fragmentShader() {
+    return `uniform vec3 colorA; 
+                    uniform vec3 colorB; 
+                    varying vec3 vUv;
+      
+                    void main() {
+                  gl_FragColor = vec4(mix(colorA, colorB, vUv.z), 1.0);
+                    }`;
+}
+
+
+export function createMaterials() {
+    const cubeShader = new THREE.ShaderMaterial({
+        uniforms: {
+            colorA: { type: "vec3", value: new THREE.Color(0xff0000) },
+            colorB: { type: "vec3", value: new THREE.Color(0x0000ff) },
+        },
+        vertexShader: vertexShader(),
+        fragmentShader: fragmentShader(),
```

Comment:
This is an interesting way of including shaders. Is this the best way to do it for threejs?

Reply:
A custom shader is a small program written in GLSL that runs on the GPU. 
A ShaderMaterial will only be rendered properly by WebGLRenderer, since the GLSL code in the vertexShader and fragmentShader properties must be compiled and run on the GPU using WebGL. 
Source: https://threejs.org/docs/#api/en/materials/ShaderMaterial

Yes, sir, since this is not evaluated in the regular way, this appears to be the best way for us to include shaders